src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -468,31 +469,33 @@ synchronized private boolean connectOne(long sid, InetSocketAddress electionAddr
      */
     
     synchronized void connectOne(long sid){
+        connectOne(sid, self.getLastSeenQuorumVerifier());
```

Comment:
Here connectOne will no longer hold same QuorumPeer for synchronization - instead the qvLock inside QuorumPeer will be used.